junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/ExtensionRegistry.java

```diff
@@ -88,28 +90,47 @@ private ExtensionRegistry(ExtensionRegistry parent) {
 	}
 
 	/**
-	 * Get all {@code Extensions} in this registry or one of its ancestors of
+	 * Stream all {@code Extensions} in this registry or one of its ancestors of
 	 * the specified type.
 	 *
 	 * @param extensionType the type of {@link Extension} to get
 	 * @see #getReversedExtensions(Class)
-	 * @see #stream(Class)
+	 * @see #getExtensions(Class)
 	 */
-	public <E extends Extension> List<E> getExtensions(Class<E> extensionType) {
-		List<E> extensions = new ArrayList<>();
-
-		if (this.parent != null) {
-			extensions.addAll(this.parent.getExtensions(extensionType));
+	public <E extends Extension> Stream<E> streamExtensions(Class<E> extensionType) {
```

Comment:
Okay, sounds good to me. :smiley:

Just so that you know where I'm coming from, the reason I disliked the idea of `streamExtensions` was because in OpenJDK 9 there is a forthcoming method `JarFile::versionedStream` (https://bugs.openjdk.java.net/browse/JDK-8163798) which to me looks and feels more like a noun than a verb and which uses "Stream" at the end of the phrase rather than the beginning, so I thought that, if you wanted to stick with something like `streamExtensions`, it would be less jarring and more consistent to use `extensionsStream`. But now that we've both decided on `getExtensionsAsStream`, which follows JUnit 5's coding style better, this is no longer a concern for me. :smile: